packages/components/nodes/vectorstores/Postgres/driver/PGVector.ts

```diff
@@ -0,0 +1,118 @@
+import { VectorStoreDriver } from './Base'
+import { FLOWISE_CHATID } from '../../../../src'
+import { DistanceStrategy, PGVectorStore, PGVectorStoreArgs } from '@langchain/community/vectorstores/pgvector'
+import { Document } from '@langchain/core/documents'
+import { PoolConfig } from 'pg'
+import { getContentColumnName } from '../utils'
+
+export class PGVectorDriver extends VectorStoreDriver {
+    static CONTENT_COLUMN_NAME_DEFAULT: string = 'pageContent'
+
+    protected _postgresConnectionOptions: PoolConfig
+
+    protected async getPostgresConnectionOptions() {
+        if (!this._postgresConnectionOptions) {
+            const { user, password } = await this.getCredentials()
+            const additionalConfig = this.nodeData.inputs?.additionalConfig as string
+
+            let additionalConfiguration = {}
+
+            if (additionalConfig) {
+                try {
+                    additionalConfiguration = typeof additionalConfig === 'object' ? additionalConfig : JSON.parse(additionalConfig)
+                } catch (exception) {
+                    throw new Error('Invalid JSON in the Additional Configuration: ' + exception)
+                }
+            }
+
+            this._postgresConnectionOptions = {
+                ...additionalConfiguration,
+                host: this.getHost(),
+                port: this.getPort(),
+                user: user,
+                password: password,
+                database: this.getDatabase()
+            }
+        }
+
+        return this._postgresConnectionOptions
+    }
+
+    async getArgs(): Promise<PGVectorStoreArgs> {
+        return {
+            postgresConnectionOptions: await this.getPostgresConnectionOptions(),
+            tableName: this.getTableName(),
+            columns: {
+                contentColumnName: getContentColumnName(this.nodeData)
+            },
+            distanceStrategy: (this.nodeData.inputs?.distanceStrategy || 'cosine') as DistanceStrategy
+        }
+    }
+
+    async instanciate(metadataFilters?: any) {
+        return this.adaptInstance(await PGVectorStore.initialize(this.getEmbeddings(), await this.getArgs()), metadataFilters)
+    }
+
+    async fromDocuments(documents: Document[]) {
+        const instance = await this.instanciate()
+
+        await instance.addDocuments(documents)
+
+        return this.adaptInstance(instance)
+    }
+
+    protected async adaptInstance(instance: PGVectorStore, metadataFilters?: any): Promise<PGVectorStore> {
+        const { [FLOWISE_CHATID]: chatId, ...pgMetadataFilter } = metadataFilters
```

Comment:
this error when `metadatFilters` is undefined

```
const { [FLOWISE_CHATID]: chatId, ...pgMetadataFilter } = metadataFilters || {}
```

Reply:
i'll fix

Reply:
Done in https://github.com/FlowiseAI/Flowise/pull/3367/commits/a7de41465d97704e2bd1aaeaabf88984b16e6e6c

I'm working on some issues of my project ATM, but i'll re-test all this stuff asap